src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -20,6 +21,16 @@ namespace System
     //       to also change the test class.
     internal static class ConsolePal
     {
+        private static readonly object s_cursorGate = new object();
+        private static int s_cursorVersion;     // Gets incremented each time the cursor position changed.
+                                                // Used to synchronize between s_cursorGate locked blocks.
+        private static int s_cursorLeft = -1;   // Cached CursorLeft, -1 when invalid.
+        private static int s_cursorTop;         // Cached CursorTop, invalid when s_cursorLeft == -1.
+        private static int s_windowWidth = -1;  // Cached WindowWidth, -1 when invalid.
+        private static int s_windowHeight = -1; // Cached WindowHeight, invalid when s_windowWidth == -1.
```

Comment:
How come this is initialized to -1 but s_cursorTop isn't?

Reply:
`s_windowHeight` can be left uninitialized too.